src/sgame/sg_main.cpp

```diff
@@ -2991,9 +2991,20 @@ void G_RunFrame( int levelTime )
 void G_PrepareEntityNetCode() {
```

Comment:
I don't like special casing the PrepareNetCode message for spectators *too much*. Can we not live with having spectators lag up to a single server frame behind? A lot of the server side gamelogic (in particular thinkers) assumes that such a lag doesn't matters.